### PR TITLE
Implement the ISNM Making a Mockery

### PR DESCRIPTION
**Pull request by jmcmorris**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

This adds the ISNM battlefield "Making a Mockery"

## Retail Proof

[https://ffxiclopedia.fandom.com/wiki/Making_a_Mockery](https://ffxiclopedia.fandom.com/wiki/Making_a_Mockery)
[Melo Captures 1 + 2](https://discord.com/channels/443544205206355968/609478860593037395/1275555506919837726)
[Melo Captures 3 + 4](https://discord.com/channels/443544205206355968/609478860593037395/1277388380866482298)

## Steps to test these changes

`!zone <Jade Sepulcher>`
`!addkeyitem 807`
Go to the door and enter the Making a Mockery battlefield
Test the battlefield according to the above retail data

### CODE REVIEW

**Review by zach2good — APPROVED** [review]
All looks reasonable to me, thanks 👍

**Review by Xaver-DaRed — APPROVED** [review]
lgtm aswell.

File: scripts/zones/Jade_Sepulcher/mobs/Mocking_Colibri.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    -- Remove dots when the mob disengages such as when charming everyone
+    mob:delStatusEffect(xi.effect.POISON)
+    mob:delStatusEffect(xi.effect.BURN)
+    mob:delStatusEffect(xi.effect.FROST)
+    mob:delStatusEffect(xi.effect.CHOKE)
+    mob:delStatusEffect(xi.effect.RASP)
+    mob:delStatusEffect(xi.effect.SHOCK)
+    mob:delStatusEffect(xi.effect.DROWN)
+    mob:delStatusEffect(xi.effect.DIA)
+    mob:delStatusEffect(xi.effect.BIO)
```
Comment (Xaver-DaRed) [inline scripts/zones/Jade_Sepulcher/mobs/Mocking_Colibri.lua:28]: This is interesting. Is this something common?

Not for this PR, but I am wondering, if this is something common, if it should be globablized

Comment (jmcmorris) [reply scripts/zones/Jade_Sepulcher/mobs/Mocking_Colibri.lua:28]: Not that I know of. I at first was using `delStatusEffectsByFlag(xi.effectFlags.ERASABLE)` but Melo tested and found that Slow would not be removed. I'm kind of surprised there isn't a DoT effect flag. I don't know if the flags are something we made up or if they are tied to retail in some way but if we can add some I'd say that'd be the way to do this. That's a whole other PR though.

**Pull request merged**